index.js

```diff
@@ -1,45 +1,90 @@
 #!/usr/bin/env node
 const { execSync } = require('child_process')
+
 const yargs = require('yargs/yargs')
+const chalk = require('chalk')
 
 // from @eqworks/release convention
 const R = /(?<cat>\S+?)(\/(?<t2>\S+))? - (?<title>.*)/
 
+const SEV_COLOR = {
+  log: 'green',
+  warn: 'yellow',
+  error: 'red',
+}
+const SEV_PRE = {
+  log: '✓',
+  warn: '⚠',
+  error: '✖',
+}
+const _log = (sev = 'log') => m => (console[sev])((chalk[SEV_COLOR[sev]])(`${SEV_PRE[sev]} ${m}`))
+const log = _log()
+const warn = _log('warn')
+const error = _log('error')
+const isGood = m => m.match(R)
+const isBad = m => !isGood(m)
+
 if (require.main === module) {
-  const { argv } = yargs(process.argv.slice(2))
-  const verbose = argv.verbose || argv.v
-  const base = argv.base || argv.b || 'origin/main'
+  const { argv } = yargs(process.argv.slice(2)).options({
+    verbose: {
+      alias: 'v',
+      description: 'verbose/debug flag',
+      type: 'boolean',
+      default: false,
+    },
+    base: {
+      alias: 'b',
+      description: 'base git ref to compare with the head ref',
+      type: 'string',
+      default: 'origin/main',
+    },
+    head: {
+      alias: 'h',
+      description: 'head git ref to compare with the base ref',
+      type: 'string',
+      default: 'HEAD',
+    },
+    fetch: {
+      description: 'perform a git fetch before commit-watch',
+      type: 'boolean',
+      default: false,
+    },
+  })
 
   if (argv.fetch) {
-    execSync('git fetch --prune')
+    execSync('git fetch')
   }
 
+  const { verbose, base, head } = argv
+
   try {
-    const messages = execSync(`git log --no-merges --format='%s' ${base}..HEAD`, { stdio: 'pipe' }).toString().trim()
+    const gitLog = `git log --no-merges --format='%s' ${base}..${head}`
+    const messages = execSync(gitLog, { stdio: 'pipe' }).toString().trim()
     if (!messages) {
       if (verbose) {
-        console.warn(`No commits between ${base}..HEAD`)
+        log(`No commits between ${base}..${head}`)
       }
       process.exit(0)
     }
     const ms = messages.split('\n')
-    const bad = ms.filter(m => !m.match(R))
+    const bad = ms.filter(isBad)
+    const good = ms.filter(isGood)
     if (bad.length) {
       if (verbose) {
-        console.error(`${bad.length}/${ms.length} commit message${bad.length > 1 ? 's' : ''} do${bad.length === 1 ? 'es' : ''} not conform to the convention ${R}\n`)
-        bad.forEach(m => console.error(m))
+        warn(`${bad.length}/${ms.length} do${bad.length === 1 ? 'es' : ''} not match RegExp${R}\n`)
```

Comment:
would recommend including `category[/sub-category] - subject title` instead of the actual RegExp in error message (like whats included in the readme) - might not be clear to someone new what the pattern looks like based on the RegExp.

Reply:
makes sense